platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -142,6 +147,42 @@ private static void initTestingMetricsReporter() {
     METER_REGISTRY.add(new SimpleMeterRegistry());
   }
 
+  private static void initPrometheusPushGatewayReporter(String serviceName,
+      int reportIntervalSec,
+      String gatewayUrlAddress) {
+    LOGGER.info("Initializing Prometheus PushGateway Reporter with urlAddress: {}, jobName: {}. "
+        + "Metric is configured get pushed for every {} seconds", gatewayUrlAddress, serviceName,
+        reportIntervalSec);
+
+    if (gatewayUrlAddress == null || gatewayUrlAddress.isEmpty()) {
+      throw new IllegalArgumentException("pushUrlAddress configuration is not specified.");
+    }
+
+    METER_REGISTRY.add(new PrometheusPushMeterRegistry(
+    new PrometheusPushRegistryConfig() {
+      @Override
+      public String jobName() {
+        return serviceName;
+      }
+
+      @Override
+      public String prefix() {
+        return "prometheus";
```

Comment:
what's this used for?

Reply:
This is to scope the meter's property in configuration. Actually making this "pushgateway" is better so that it's not conflicting to premethues